apps/server/src/modules/authorization-rules/rules/user.rule.ts

```diff
@@ -12,15 +13,20 @@ export class UserRule implements Rule<User> {
 	}
 
 	public isApplicable(user: User, object: unknown): boolean {
-		const isMatched = object instanceof User;
+		const isMatched = object instanceof UserDo;
 
 		return isMatched;
 	}
 
-	public hasPermission(user: User, entity: User, context: AuthorizationContext): boolean {
+	public hasPermission(user: User, entity: UserDo, context: AuthorizationContext): boolean {
 		const hasPermission = this.authorizationHelper.hasAllPermissions(user, context.requiredPermissions);
-		const isOwner = user === entity;
 
-		return hasPermission || isOwner;
+		const isOwner = user.id === entity.id;
```

Comment:
isOwner is not technically false, but the code is more readable if it was isHimself